### PR TITLE
Resetdigital bid Adapter: add fields and suport for coppa

### PR DESCRIPTION
**Pull request by nicolasresetdigital**
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
add new field to build request 
add coppa and gdpr support

feel free to contact me here: nicolas@resetdigital.co

### CODE REVIEW

File: modules/resetdigitalBidAdapter.js [hunk 1]
```diff
@@ -43,16 +43,29 @@ export const spec = {
       };
     }
 
+    if (bidderRequest && bidderRequest.uspConsent) {
+      payload.ccpa = bidderRequest.uspConsent;
```
Comment (patmmccann) [inline modules/resetdigitalBidAdapter.js:47]: Your title says gdpr but your code doesn't?

File: modules/resetdigitalBidAdapter.js [hunk 2]
```diff
@@ -43,16 +43,29 @@ export const spec = {
...
+
     for (let x = 0; x < validBidRequests.length; x++) {
       let req = validBidRequests[x]
 
       payload.imps.push({
         pub_id: req.params.pubId,
+        site_id: req.params.siteID ? req.params.siteID : null,
+        placement_id: req.params.placement ? req.params.placement : null,
+        position: req.params.position ? req.params.position : null,
+        bid_floor: req.params.bidFloor ? req.params.bidFloor : null,
```
Comment (patmmccann) [inline modules/resetdigitalBidAdapter.js:58]: You must also support the mostly

File: modules/resetdigitalBidAdapter.js (hunk 1 above)
Comment (nicolasresetdigital) [reply modules/resetdigitalBidAdapter.js:47]: gdpr it's on line 39

File: modules/resetdigitalBidAdapter.js (hunk 2 above)
Comment (nicolasresetdigital) [reply modules/resetdigitalBidAdapter.js:58]: Hi, can you give me an example or a link?  I don't know what you mean by the module, I appreciate any help thanks

**Comment by nicolasresetdigital** [comment]
Hi team, any news about this? cheers

File: modules/resetdigitalBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/resetdigitalBidAdapter.js:58]: sure, if you take a floor on params, you must also take one in the bidrequest object. Here is a good example https://github.com/prebid/Prebid.js/pull/9279/files

File: modules/resetdigitalBidAdapter.js [hunk 3]
```diff
@@ -43,16 +43,29 @@ export const spec = {
...
+        lat_long: req.params.latLong ? req.params.latLong : null,
+        inventory: req.params.inventory ? req.params.inventory : null,
+        visitor: req.params.visitor ? req.params.visitor : null,
+        keywords: req.params.keywords ? req.params.keywords : null,
```
Comment (patmmccann) [inline modules/resetdigitalBidAdapter.js:62]: this is also true about keywords, the bid request object should have keywords you can append to the keywords you see in params

File: modules/resetdigitalBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/resetdigitalBidAdapter.js:47]: right but that is already there, it isnt changing.

File: modules/resetdigitalBidAdapter.js (hunk 2 above)
Comment (nicolasresetdigital) [reply modules/resetdigitalBidAdapter.js:58]: good! that really helped me, thanks!

Comment (nicolasresetdigital) [reply modules/resetdigitalBidAdapter.js]: fixed

File: modules/resetdigitalBidAdapter.js (hunk 3 above)
Comment (nicolasresetdigital) [reply modules/resetdigitalBidAdapter.js:81]: I reviewed other adapters and couldn't find one that takes the keywords from the bid request object, so I think we are covered about that. in another case can you share an example with me?

**Comment by nicolasresetdigital** [comment]
Hi @patmmccann I think this could be a final version, please let me know what you think, and very thanks for your help

**Comment by ChrisHuie** [comment]
@nicolasresetdigital can you pull in master? Hopefully fixes the build error

**Comment by nicolasresetdigital** [comment]
Thanks, @ChrisHuie I fixed the building error, @patmmccann I fixed keywords comments, let me know if it needs any other changes, cheers

**Review by patmmccann — APPROVED** [review]

**Pull request merged**